var/spack/repos/builtin/packages/caliper/package.py

```diff
@@ -71,7 +74,7 @@ def cmake_args(self):
             '-DWITH_SOSFLOW=%s'  % ('On' if '+sosflow'  in spec else 'Off'),
             '-DWITH_SAMPLER=%s'  % ('On' if '+sampler'  in spec else 'Off'),
             '-DWITH_MPI=%s'      % ('On' if '+mpi'      in spec else 'Off'),
-            '-DWITH_MPIT=%s' % ('On' if spec.satisfies('^mpi@3:') else 'Off')
+            '-DWITH_MPIT=%s'     % ('On' if '+mpit'     in spec else 'Off')
```

Comment:
does `+mpit` conflict with `mpi@:2.9`? If so add this with a `conflicts` statement

Reply:
Not sure how to code this. When I write `conflicts('+mpit', when='^mpi@:2.9')` it still marks a conflict with MPI 3 implementations. I guess this is because MPI 3 packages also provide mpi@2? I somehow need to encode "is not MPI 3" rather than "is MPI 2.x" in the when clause.

Reply:
Ah I see, maybe @alalazo knows more about the inner workings of `mpi`

Reply:
I think it would make more sense to say `depends_on('mpi@3:', when='+mpit')`

Reply:
Good point @scheibelp . I'll do that.